src/odemis/driver/xt_client.py

```diff
@@ -2404,40 +2400,31 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 axes_to_check = linear_axes_to_check | rotational_axes_to_check
 
                 while True:
-                    current_pos = self.parent.get_stage_position()
-                    # Every axis requested to move should have moved (compared to the position before starting)
-                    # if there is some change w.r.t starting position, proceed to check the target position
-                    # Note that we don't use almost_equal() on the floats, because all we care about is whether
-                    # the position has changed. Even a very tiny change is a sign we've received a position update.
-                    if all(current_pos[a] != op for a, op in orig_pos.items()):
-
-                        if axes_to_check:
-                            axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
-                                                          axes=axes_to_check, rot_axes=rotational_axes_to_check,
-                                                          atol_linear=tol_linear, atol_rotation=tol_rotation)
-                        else:
-                            axes_updated = True  # No axis to check => move is done
+                    if not axes_to_check:
+                        break
```

Comment:
Maybe we can put this as a condition in while, as while axes_to_check: instead of while True: